src/components/DashBoard.js

```diff
@@ -69,6 +83,14 @@ const DashBoard = ({ showFullMap, showSiteDetail, backButtonHandler, siteDetail,
     </ThemeProvider>
   );
 
+  const fullMapToggle = (
+    <ThemeProvider theme={fullZoomTheme}>
```

Comment:
I would place the ThemeProvider at a higher level of the component tree, instead of having multiple themes. The purpose of the themeProvider is to have one source of truth, instead of multiple. 

It's worth noting that you don't need ThemeProvider over the same react component. So long as it's at a higher level, the context will carry itself all the way to the bottom.